src/index.js

```diff
@@ -94,6 +96,24 @@ class BlockService {
     return this._repo.blocks.get(cid, callback)
   }
 
+  /**
+   * Get multiple blocks a block by cid.
+   *
+   * @param {Array<CID>} cids
+   * @param {function(Error, Block)} callback
+   * @returns {void}
+   */
+  getMany (cids, callback) {
+    if (!Array.isArray(cids)) {
+      return callback(new Error('first arg must be an array of cids'))
+    }
+    if (this.hasExchange()) {
+      return this._bitswap.getMany(cids, callback)
```

Comment:
One thing that baffled me at first sight was that this synchronously returns, while in the next few lines, we asynchronously fetch from the local repository.

After looking at Bitswap's API I noticed that `getMany()` isn't actually synchronous either, which makes me wonder:

Are we only **returning** here to stop `getMany()` ?

If yes, couldn't we then also go with:

```
this.hasExchange() ? 
  this._bitswap.getMany(cids, callback) : 
  asyncMap(cids, (cid, cb) => this._repo.blocks.get(cid, db, callback)
```

Then it might be more obvious that we aren't interested in returning anything here (even though **get**Many kind of says the opposite, but hey... it's async)

Reply:
Got it. Yea I don't have strong feelings about ternary vs none-ternary either.

My concern was mainly that it looks like we're actually returning something synchronously, while the API is async.

Reply:
@PascalPrecht I agree that it is confusing to someone not used to that style. Perhaps early returns are more clear when written as:

```
if (this.hasExchange()) {
  this._bitswap.getMany(cids, callback)
  return
}
```

Though I don't have strong feelings about it. Either way is fine for me, but it should perhaps end up in the [Coding Guideles](https://github.com/ipfs/community/blob/master/js-code-guidelines.md) to make it easy to get started.